x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -165,11 +165,17 @@ protected LogicalPlan rule(Filter filter) {
                 Expression result = cmp;
                 // expr == null || expr != null
                 if (cmp instanceof Equals || cmp instanceof NotEquals) {
+                    Expression comparableToNull = null;
                     if (cmp.right().foldable() && cmp.right().fold() == null) {
+                        comparableToNull = cmp.left();
+                    } else if (cmp.left().foldable() && cmp.left().fold() == null) {
+                        comparableToNull = cmp.right();
+                    }
+                    if (comparableToNull != null) {
```

Comment:
Since the left side is now considered for null, do we have tests to check this scenario?

Reply:
@costin in OptimizerTests we have [`testIsNull` and `testIsNotNull` that both handle `null` on the right and left hand side of the expression](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/OptimizerTests.java#L97-L133). Do you think those are enough?